django-stubs/views/generic/detail.pyi

```diff
@@ -7,7 +7,7 @@ from django.views.generic.base import ContextMixin, TemplateResponseMixin, View
 _M = TypeVar("_M", bound=models.Model)
 
 class SingleObjectMixin(Generic[_M], ContextMixin):
-    model: type[_M]
+    model: type[_M] | None
```

Comment:
This can be a breaking change for a lot of code.

Reply:
Yeah perhaps. Its default value is `None` though. So if you're declaring a view, subclassing `SingleObjectMixin` and also declare `model`. The type will be that model, `None` would go away.

Do you think we should add that as a test?

Any other, more generic, code expecting this kind of view should hopefully be a bit happy that `None` is covered.

Reply:
Yes, sounds good.

Reply:
> [...] So if you're declaring a view, subclassing SingleObjectMixin and also declare model. The type will be that model, None would go away. [...]

Well, so I thought, `pyright` behaves differently in regards to this and keeps the original type 🙂

Reply:
I think that we have to decide: practicality or purity. I think that the first one is more important in this case. `None` is just a default / sentinel value.

Reply:
Sure, I don't have that strong opinions about the type here but I would consider it fairly easy to run in to an `AttributeError: 'NoneType' object has no attribute 'model'`

Reply:
But if it was a larger issue, someone would've probably reported it by now. So it should probably work out just fine